crates/lib3h_protocol/src/error.rs

```diff
@@ -0,0 +1,105 @@
+//! Lib3h_protocol custom error definition.
+
+use std::{error::Error as StdError, fmt, io, result};
+// use serde::Deserializer::Error as DerserializerError;
+use serde::de::value::Error as DeserializeError;
+
+/// A type alias for `Result<T, Lib3hProtocolError>`.
+pub type Lib3hProtocolResult<T> = result::Result<T, Lib3hProtocolError>;
+
+/// An error that can occur when interacting with the algorithm.
+#[derive(Debug)]
+pub struct Lib3hProtocolError(Box<ErrorKind>);
+
+impl Lib3hProtocolError {
+    /// A crate private constructor for `Error`.
+    pub fn new(kind: ErrorKind) -> Lib3hProtocolError {
+        Lib3hProtocolError(Box::new(kind))
+    }
+    /// Return the specific type of this error.
+    pub fn kind(&self) -> &ErrorKind {
+        &self.0
+    }
+
+    /// Unwrap this error into its underlying type.
+    pub fn into_kind(self) -> ErrorKind {
+        *self.0
+    }
+}
+
+/// The specific type of an error.
+#[derive(Debug)]
+pub enum ErrorKind {
+    /// An I/O error that occurred while processing a data stream.
+    Io(io::Error),
+    /// Error occuring when using `transport`.
+    TransportError(String),
+    /// An error occuring whiling trying to deserialize stuff during gossiping for example.
+    // DeserializeError(String),
+    DeserializeError(DeserializeError),
+    /// Yet undefined error.
+    Other(String),
```

Comment:
Is this common naming? In our org, I've seen this called a couple things, including `Generic(String),`. If there's a rust-world standard naming, I'm cool with it, or if not, we should try to pick something and stick with it.

Reply:
The standard naming way is indeed to use [Other](https://doc.rust-lang.org/std/io/enum.ErrorKind.html#variant.Other).
But I'm not too picky about this either :)

Reply:
I'm fine with `Other`.